### PR TITLE
Marco

### PR DESCRIPTION
**Pull request by MarcoSchaefer**
Merging...

### CODE REVIEW

File: Syntax.jj [hunk 1]
```diff
@@ -1,41 +1,79 @@
...
+    TOKEN: {<ASK: "?">}
+    TOKEN: {<GOE: ">=">}
+    TOKEN: {<LOE: "<=">}
+    TOKEN: {<EQUAL: "=">}
+    TOKEN: {<EQUALS: "==">}
+    TOKEN: {<COLON: ":">}
+    TOKEN: {<COMA: ",">}
+    TOKEN: {<AND: "&">}
+    TOKEN: {<DIV: "/">}
+    /*Programa e Bloco*/
```
Comment (abbudao) [inline Syntax.jj:72]: Should be an inline comment, not a block comment.

File: Syntax.jj [hunk 2]
```diff
@@ -47,9 +85,9 @@ PARSER_END(Adder)
     void Block():
     {}
     {
-        <BEGIN>Statements()<END>
+        <BEGIN>[Statements()]<END>
     }
-
+    /*Declaracoes*/
```
Comment (abbudao) [inline Syntax.jj:90]: Should be an inline comment

File: Syntax.jj [hunk 3]
```diff
@@ -1,41 +1,79 @@
...
+    TOKEN: {<PROC: "procedure">}
+    TOKEN: {<PARAM: "parameters">}
+    TOKEN: {<VAR: "var">}
+    TOKEN: {<FUNCTION: "function">}
+    TOKEN: {<RETURNS: "returns">}
+    TOKEN: {<RETURN: "return">}
+    TOKEN: {<THEN: "then">}
+    TOKEN: {<ARRAY: "array">}
+    TOKEN: {<EOL: ";">}
+    TOKEN: {<IDENTIFIER: ("_")* (["a"-"z", "A"-"Z", "_"])+ (<NUM>)*>}
```
Comment (abbudao) [inline Syntax.jj:49]: I guess this isn't the proposed behavior. 
`_8123123` should be an okay variable name, but in this way it is not accepted.
What we want is something starting with a letter or a `"_"` and then anything valid after.
Something like this:   
`{<IDENTIFIER: (["a"-"z", "A"-"Z", "_"]) ["a"-"z", "A"-"Z", "_",<NUM>] + >}`

Ps: Not sure if writting `<NUM>` inside the `[]` is valid.

File: Syntax.jj [hunk 4]
```diff
@@ -94,23 +132,202 @@ PARSER_END(Adder)
...
+    {
+        <FUNCTION> <IDENTIFIER> <RETURNS> DataType() <EOL>
+    }
+
+    void FunctionBlock():
+    {}
+    {
+        <BEGIN> [Statements()] ReturnStatement() <END>
+    }
+    /* Comandos */
```
Comment (abbudao) [inline Syntax.jj:216]: Change to Inline comment

**Review by abbudao — APPROVED** [review]

File: Syntax.jj (hunk 3 above)
Comment (igmarinelli) [reply Syntax.jj:49]: Parabéns aos envolvidos

**Pull request merged**